lib_vehicle_model/src/lib_vehicle_model/ConstraintChecker.cpp

```diff
@@ -58,18 +58,8 @@ ConstraintChecker::ConstraintChecker(std::shared_ptr<ParameterServer> parameter_
 void ConstraintChecker::validateInitialState(const VehicleState& initial_state) {
   std::ostringstream msg;
 
-  if (abs(initial_state.x_vel) > max_forward_speed_) {
-    msg << "Invalid initial_state with x_vel: " << initial_state.x_vel << "  above max speed of: " << max_forward_speed_;
-    throw std::invalid_argument(msg.str());
-  }
-
-  if (abs(initial_state.y_vel) > max_forward_speed_) {
-    msg << "Invalid initial_state with y_vel: " << initial_state.y_vel << " above max speed of: " << max_forward_speed_;
-    throw std::invalid_argument(msg.str());
-  }
-
-  if (initial_state.steering_angle < min_steering_angle_) {
-    msg << "Invalid initial_state with steering angle: " << initial_state.steering_angle << " is below min of: " << min_steering_angle_;
+  if ((abs(initial_state.x_vel) + abs(initial_state.y_vel)) > max_forward_speed_) {
```

Comment:
I think it would based on vehicle heading. Basically it is the projection of the sum of those two vectors in heading direction.

Reply:
My mistake. The magnitude of velocity vector is what we want. So sqrt(x_val^2 + y_vel^2). I wish there was a way to not have the square root though

Reply:
I am wondering if we should remove this parameter. It is the only constraint which is not tied to the configuration of the vehicle or bounds on control inputs.